modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/semantic/SemanticTermsEnricher.java

```diff
@@ -0,0 +1,96 @@
+import org.mule.runtime.api.meta.model.declaration.fluent.SourceDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.WithSemanticTermsDeclaration;
+import org.mule.runtime.extension.api.declaration.fluent.util.IdempotentDeclarationWalker;
+import org.mule.runtime.extension.api.declaration.type.ExtensionsTypeLoaderFactory;
+import org.mule.runtime.extension.api.loader.ExtensionLoadingContext;
+import org.mule.runtime.module.extension.api.loader.java.type.WithAnnotations;
+import org.mule.runtime.module.extension.internal.loader.enricher.AbstractAnnotatedDeclarationEnricher;
+import org.mule.runtime.module.extension.internal.loader.java.type.runtime.MethodWrapper;
+
+import java.util.Set;
+
+/**
+ * An enricher which adds semantic terms based on annotations
+ *
+ * @since 4.4.0ExtensionDocumentationResourceGenerator.java
+ */
+public class SemanticTermsEnricher extends AbstractAnnotatedDeclarationEnricher {
+
+  private ClassTypeLoader typeLoader = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader();
+
+  @Override
+  public void enrich(ExtensionLoadingContext extensionLoadingContext) {
+    new IdempotentDeclarationWalker() {
+
+      @Override
+      protected void onConnectionProvider(ConnectionProviderDeclaration declaration) {
+        extractType(declaration).ifPresent(type -> addSemanticTerms(declaration, type));
+      }
+
+      @Override
+      protected void onOperation(OperationDeclaration declaration) {
+        extractImplementingMethod(declaration)
+            .map(method -> new MethodWrapper(method, typeLoader))
+            .ifPresent(method -> addSemanticTerms(declaration, method));
+      }
+
+      @Override
+      protected void onSource(SourceDeclaration declaration) {
+        extractType(declaration).ifPresent(type -> addSemanticTerms(declaration, type));
+      }
+
+      @Override
+      protected void onParameter(ParameterGroupDeclaration parameterGroup, ParameterDeclaration parameter) {
+        extractDeclaredParameter(parameter).ifPresent(e -> addSemanticTerms(parameter, e));
+        Set<String> typeTerms = getSemanticTerms(parameter.getType());
+
+        addTermIfPresent(typeTerms, parameter, PROXY_CONFIGURATION_TYPE, PROXY_CONFIGURATION_PARAMETER);
+        addTermIfPresent(typeTerms, parameter, NTLM_PROXY_CONFIGURATION, NTLM_PROXY_CONFIGURATION_PARAMETER);
+        if (typeTerms.contains(SECRET)) {
+          parameter.getType().accept(new BasicTypeMetadataVisitor() {
+
+            @Override
+            protected void visitBasicType(MetadataType metadataType) {
+              typeTerms.remove(SECRET);
+              typeTerms.add(SCALAR_SECRET);
+            }
+          });
+        }
+      }
+    }.walk(extensionLoadingContext.getExtensionDeclarer().getDeclaration());
+  }
+
+  private void addTermIfPresent(Set<String> terms, WithSemanticTermsDeclaration declaration, String term, String mappedTerm) {
```

Comment:
Do we want to maintain both terms in the set? for example NTLM_PROXY_CONFIGURATION and NTLM_PROXY_CONFIGURATION_PARAMETER, or should it be replaced?

Reply:
`NTLM_PROXY_CONFIGURATION` is kept as a type term, while `NTLM_PROXY_CONFIGURATION_PARAMETER` gets added on the parameter